### DISCUSSION TITLE
error in plotting LIS output

### DISCUSSION THREAD

**Discussion by hashmifatima** [NASA-LIS/LISF#1303] [General]
Hi

I am running a GDAS METforcing case with Noah.3.3 for 2 days. I have written output in "2d gridspace" netcdf format.  I have got files like this:
1. LIS_RST_NOAH33_200210310000.d01.nc
2. LIS_RST_NOAH33_200210310100.d01.nc
3. LIS_HIST_200210310000.d01.nc

I trying to plot these files by xdfopen.

DSET       ^LIS_HIST_200210310000.d01.nc
DTYPE netcdf
UNDEF       -9999.
TITLE lis
XDEF    east_west 1440   LINEAR  -179.875    0.25
YDEF  north_south  600   LINEAR   -59.875    0.25
ZDEF SoilMoist_profiles 4 levels 0.1 0.4 1.0 2.0
TDEF time 1 LINEAR 00Z31oct2002 1dy
VARS         11
SnowDepth     1  99  ** Snow depth [m]
SoilMoist1    1  99  ** Soil water content layer 1 [kg m-2]
SoilMoist2    1  99  ** Soil water content layer 2 [kg m-2]
SoilMoist3    1  99  ** Soil water content layer 3 [kg m-2]
SoilMoist4    1  99  ** Soil water content layer 4 [kg m-2]
SoilTemp1     1  99  ** Soil temperature layer 1 [K]
SoilTemp2     1  99  ** Soil temperature layer 2 [K]
SoilTemp3     1  99  ** Soil temperature layer 3 [K]
SoilTemp4     1  99  ** Soil temperature layer 4 [K]
CanopInt      1  99  ** Canopy interception [kg m-2]
Snowcover     1  99  ** Snow cover fraction [0-1]
ENDVARS

 I am getting this error: 

ga-> xdfopen output.xdf
Scanning Descriptor File:  output.xdf
gadsdf: SDF file does not have any non-coordinate variables.
SDF Descriptor file output.xdf was not successfully opened & parsed.
 
~Hashmi

**Comment by dmocko** [comment]
Hi @hashmifatima 

Please try your GrADS .xdf file without all of the lines from/to "VAR" and "ENDVARS" - including those two lines.

In other words, just end your .xdf file with the TDEF line.  GrADS will figure out the variables in your netCDF file.

For more info on how to build a .xdf file, please see this GrADS documentation: http://cola.gmu.edu/grads/gadoc/SDFdescriptorfile.html

After you successfully open your GrADS file this way, you can type `q ctlinfo` at the GrADS prompt to see the list of variables.

Here is an example from one of my "2d gridspace" LIS_HIST netCDF files opened with GrADS:

```
ga-> q ctlinfo                            
dset LIS_HIST_202007010000.d01.nc
title Noah-3.9 LSM
undef -9999
dtype netcdf
xdef 2560 linear -179.93 0.140625
ydef 1920 linear -89.9531 0.09375
zdef 4 linear 1 1
tdef 1 linear 00Z01JUL2020 1440mn
vars 22
lat=>lat  0  y,x  latitude
lon=>lon  0  y,x  longitude
Swnet_tavg=>swnet_tavg  0  y,x  net downward shortwave radiation
Lwnet_tavg=>lwnet_tavg  0  y,x  net downward longwave radiation
Qle_tavg=>qle_tavg  0  y,x  latent heat flux
Qh_tavg=>qh_tavg  0  y,x  sensible heat flux
Snowf_tavg=>snowf_tavg  0  y,x  snowfall rate
Rainf_tavg=>rainf_tavg  0  y,x  precipitation rate
Evap_tavg=>evap_tavg  0  y,x  total evapotranspiration
Qs_tavg=>qs_tavg  0  y,x  surface runoff
Qsb_tavg=>qsb_tavg  0  y,x  subsurface runoff amount
AvgSurfT_tavg=>avgsurft_tavg  0  y,x  surface temperature
Albedo_tavg=>albedo_tavg  0  y,x  surface albedo
SWE_tavg=>swe_tavg  0  y,x  snow water equivalent
SnowDepth_inst=>snowdepth_inst  0  y,x  snow depth
SoilMoist_tavg=>soilmoist_tavg  4  z,y,x  soil moisture content
ECanop_tavg=>ecanop_tavg  0  y,x  interception evaporation
TVeg_tavg=>tveg_tavg  0  y,x  vegetation transpiration
ESoil_tavg=>esoil_tavg  0  y,x  bare soil evaporation
CanopInt_tavg=>canopint_tavg  0  y,x  total canopy water storage
SubSnow_tavg=>subsnow_tavg  0  y,x  snow sublimation
Snowcover_tavg=>snowcover_tavg  0  y,x  snow cover
endvars
```

You can then copy and paste the lines between/including "var" and "endvars" into your .xdf file, and try to xdfopen again.
The only reason you might need to do so, is that you can change the variable name in GrADS.  For example:

Snowcover_tavg=>snowcover  0  y,x  snow cover

This will allow you to `"d snowcover"` in GrADS, and it will read the "Snowcover_tavg" variable in the netCDF file.

Please let me know that this will work for you.

**Reply by hashmifatima** [reply]
Thanks. It works!!

~Hashmi
